event-handlers/src/test/java/no/sikt/nva/nvi/events/evaluator/EvaluateNviCandidateHandlerTest.java

```diff
@@ -523,6 +553,203 @@ void shouldCreateNewCandidateEventWhenAffiliationAreNviInstitutions() throws IOE
+                List.of(
+                        new UnverifiedNviCreator(
+                                unverifiedContributor.contributorName(),
+                                unverifiedContributor.affiliationIds()));
+
+        var expectedCandidate =
+                getDefaultExpectedCandidateBuilder()
+                        .withPublicationId(publication.id())
+                        .withPublicationBucketUri(fileUri)
+                        .withCreatorShareCount(2)
+                        .withInstitutionPoints(expectedPointsPerInstitution)
+                        .withTotalPoints(expectedTotalPoints)
+                        .withUnverifiedNviCreators(unverifiedNviCreators)
+                        .build();
+        var expectedEvaluatedMessage =
+                CandidateEvaluatedMessage.builder().withCandidateType(expectedCandidate).build();
+
+        // Call handler and parse response
+        var event = createEvent(new PersistedResourceMessage(fileUri));
+        handler.handleRequest(event, context);
+        var messageBody = getMessageBody();
+
+        // Assert that response matches exactly what we expect
+        assertEquals(expectedEvaluatedMessage, messageBody);
+    }
+
+    @Test
+    void shouldIdentifyCandidateWithMissingCountryCode() throws IOException {
+        // Define input data and mock external services
+        var affiliation =
+                ExpandedAffiliation.from(DEFAULT_SUBUNIT_AFFILIATION).withCountryCode(null).build();
+        var contributor =
+                ExpandedContributor.from(DEFAULT_VERIFIED_CONTRIBUTOR)
+                                   .withAffiliations(List.of(affiliation))
+                                   .build();
+        var publication =
+                ExpandedInputResourceGenerator.from(DEFAULT_INPUT_PUBLICATION)
+                        .withContributors(List.of(contributor))
+                        .build();
+        var fileUri =
+                s3Driver.insertFile(
+                        UnixPath.of(publication.identifier().toString()),
+                        publication.toJsonString());
+        mockCristinResponseAndCustomerApiResponseForNviInstitution(okResponse);
+
+        // Define expected response
+        var expectedCandidate =
+                getDefaultExpectedCandidateBuilder()
+                        .withPublicationId(publication.id())
+                        .withPublicationBucketUri(fileUri)
+                        .build();
+        var expectedEvaluatedMessage =
+                CandidateEvaluatedMessage.builder().withCandidateType(expectedCandidate).build();
+
+        // Call handler and parse response
+        var event = createEvent(new PersistedResourceMessage(fileUri));
+        handler.handleRequest(event, context);
+        var messageBody = getMessageBody();
+
+        // Assert that response matches exactly what we expect
+        assertEquals(expectedEvaluatedMessage, messageBody);
+    }
```

Comment:
General comment regarding this new test methods, I think they can be shortened a lot by extracting duplicated test setup to methods. Do we need comments about different sections in the test?

Reply:
Agreed, I'll try. I need the comments for readability when the test is this long, but it shouldn't be this long.